File: algorithms/1688.count-of-matches-in-tournament/solution.go
func numberOfMatches(n int) int {
	count := 0

	for n > 1 {
		if n%2 == 0 {
			n = n / 2
			count += n
		} else {
			n = (n - 1) / 2
			count += n + 1
		}
	}

	return count
}

File: algorithms/1688.count-of-matches-in-tournament/solution.rs
impl Solution {
    pub fn number_of_matches(n: i32) -> i32 {
        let mut n = n;
        let mut count: i32 = 0;

        while n > 1 {
            if n % 2 == 0 {
                n = n / 2;
                count += n;
            } else {
                n = (n - 1) / 2;
                count += n + 1;
            }
        }

        count
    }
}
